src/Servers/Kestrel/Transport.Sockets/src/SocketTransportOptions.cs

```diff
@@ -44,6 +44,14 @@ public class SocketTransportOptions
 
         public long? MaxWriteBufferSize { get; set; } = 64 * 1024;
 
+        /// <summary>
+        /// Inline application and transport continuations instead of dispatching to the threadpool.
+        /// </summary>
+        /// <remarks>
+        /// This will run application code on the IO thread which is why this is unsafe.
+        /// </remarks>
+        public bool UnsafeInlineScheduling { get; set; }
```

Comment:
RunOnTransportThread

Reply:
Doesn't sound unsafe enough!

Reply:
UnsafeRunOnTransportThread!

Reply:
```suggestion
        public bool AllowUnsafeInlineScheduling { get; set; }
```

Reply:
- We add the API and get the API added in dotnet/runtime. Setting our API should set both.
- We add the API and We use private reflection to set it on the socket itself. Setting our API should set both.
- We don't expose a public API and we also consume the env variable. (and we have an opt-out for weird cases)

Reply:
Looks like the env variable in Runtime globally affects some settings and even if you use private reflection to set the per Socket setting you don't affect the other things the env variable does. This results in massive perf regressions, i.e. 100k RPS without the env variable but with private reflection vs. 1.2M RPS with the env variable and no private reflection.